app/bundles/LeadBundle/Controller/FrequencyRuleTrait.php

```diff
@@ -60,6 +60,17 @@ protected function getFrequencyRuleForm($lead, &$viewParameters = [], &$data = n
             ]
         );
 
+        //find the email
+        $currentChannelId = null;
+        if ($viewParameters['idHash']) {
```

Comment:
Are you sure the `idHash` property always exists in the array? Can we enclose it in `!empty()` to make sure it won't throw an error if it won't exist?